tools/Dashboard/DevHome.Dashboard/Views/AddWidgetDialog.xaml.cs

```diff
@@ -20,29 +21,28 @@
 using WinUIEx;
 
 namespace DevHome.Dashboard.Views;
+
 public sealed partial class AddWidgetDialog : ContentDialog
 {
-    private readonly WidgetHost _widgetHost;
-    private readonly WidgetCatalog _widgetCatalog;
     private Widget _currentWidget;
     private static DispatcherQueue _dispatcher;
 
     public Widget AddedWidget { get; set; }
 
     public WidgetViewModel ViewModel { get; set; }
 
+    private readonly WidgetHostingService _hostingService;
+
     public AddWidgetDialog(
-        WidgetHost host,
-        WidgetCatalog catalog,
         AdaptiveCardRenderer renderer,
         DispatcherQueue dispatcher,
         ElementTheme theme)
     {
         ViewModel = new WidgetViewModel(null, Microsoft.Windows.Widgets.WidgetSize.Large, null, renderer, dispatcher);
+        _hostingService = Application.Current.GetService<WidgetHostingService>();

```

Comment:
NIT: Lookup how the SetupTool uses IHost and use IHost.GetService.  `Application.Current` breaks in unit tests.  Not a big deal right now, but something to keep in mind.

Reply:
We should avoid `Application.Current.GetService` for code that will be unit tested. However this file will not be unit tested, instead should be UI tested (`xaml.cs`)

Reply:
@dhoehna and @AmelBawa-msft , since this is part of a larger change, I'm taking note of each of your comments on this PR and will incorporate them at some point. Thanks for reviewing!